lib/evaluate_board.rb

```diff
@@ -16,14 +16,22 @@ def initialize(board_gateway)
   def execute()
     return 'Player one wins' if win_for_player?('X')
     return 'Player two wins' if win_for_player?('O')
-    board = @board_gateway.fetch_board
-    return 'Game over' if board.compact.empty?
+    return 'Game over' if game_over?
 
     ''
   end
 
   private
 
+  def game_over?
+    board = @board_gateway.fetch_board
+    if board.nil?
+      nil
+    else
+      return true if board.compact.length == 9
```

Comment:
I'd clean up these inconsistent return points.

You could write this as a guard statement

Reply:
Resolved.